pydantic/env_settings.py

```diff
@@ -144,20 +146,14 @@ def __call__(self, settings: BaseSettings) -> Dict[str, Any]:
         """
         d: Dict[str, Optional[str]] = {}
 
-        if settings.__config__.case_sensitive:
-            env_vars: Mapping[str, Optional[str]] = os.environ
-        else:
-            env_vars = {k.lower(): v for k, v in os.environ.items()}
-
-        if self.env_file is not None:
-            env_path = Path(self.env_file).expanduser()
-            if env_path.is_file():
-                env_vars = {
-                    **read_env_file(
-                        env_path, encoding=self.env_file_encoding, case_sensitive=settings.__config__.case_sensitive
-                    ),
-                    **env_vars,
-                }
+        case_sensitive = settings.__config__.case_sensitive
+        process_env_vars = os.environ if case_sensitive else {k.lower(): v for k, v in os.environ.items()}
+        dotenv_vars = (
+            {}
+            if self.env_file is None
+            else _read_env_files(self.env_file, encoding=self.env_file_encoding, case_sensitive=case_sensitive)
```

Comment:
surely make this a method on the class, then you don't need to pass `self.env_file_encoding`.

Reply:
OK. This is considering splitting dotenv processing into a whole new class.
But we decided to do it in v2, so I'll change it again.